uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -335,11 +339,13 @@ async def asgi_receive(
 
         await self.handshake_completed_event.wait()
 
-        if self.closed_event.is_set():
-            # If client disconnected, use WebSocketServerProtocol.close_code property.
+        if self.lost_connection_before_handshake:
```

Comment:
Instead of `lost_connection_before_handshake`, I can create an attribute called `lost_connection`, and do the conditional before the `await self.handshake_completed_event.wait()` above. What do you prefer? :thinking: 

(I think this alternative will work, I'm not sure if I'm missing something...)

Reply:
Meaning we wouldn’t hit `await self.handshake_completed_event.wait()` at all in connection is lost?

If that’s the case, it seems better 🤔

Reply:
What I said doesn't work.